src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs

```diff
@@ -556,6 +557,81 @@ public void Add(NamespaceOrTypeSymbol symbol)
 
                 return result;
             }
+
+            public void AddInjectedSymbols(SourceNamespaceSymbol containingNamespace)
+            {
+                const string CodeAnalysis = "CodeAnalysis";
+                const string System = "System";
+                const string Microsoft = "Microsoft";
+                const string Runtime = "Runtime";
+                const string CompilerServices = "CompilerServices";
+
+                if (containingNamespace.IsGlobalNamespace)
+                {
+                    if (!_dictionary.ContainsKey(System))
+                    {
+                        Add(makeEmptyNamespace(System));
+                    }
+                    if (!_dictionary.ContainsKey(Microsoft))
+                    {
+                        Add(makeEmptyNamespace(Microsoft));
+                    }
+                }
+                else if (isSystem(containingNamespace) && !_dictionary.ContainsKey(Runtime))
```

Comment:
>&& !_dictionary.ContainsKey(Runtime) [](start = 55, length = 36)

Consider moving inside `else if`, so we don't need to check the other `else if` cases below. Same for other `else if` conditions.